python/whylogs/api/store/local_store.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import os
+from datetime import datetime
+from functools import reduce
+from glob import glob
+from typing import Optional
+
+import whylogs as why
+from whylogs.api.store.profile_store import ProfileStore
+from whylogs.api.store.query import DateQuery
+from whylogs.api.store.utils.random_strings import random_string
+from whylogs.api.writer.local import LocalWriter
+from whylogs.core import DatasetProfileView
+
+DEFAULT_DIR = "profile_store/"
+logger = logging.getLogger(__name__)
+
+# TODO how to prevent users from creating a huge list?
+# should the profile store be aware of its min/max dates?
+
+
```

Comment:
I think we can get around this by avoiding loading everything into memory at once if the concern is just not letting the host go oom. To do that, we'd have to have make it possible for the store to use that yield iterator in the reduce.

Reply:
it's using it already :D

Reply:
the only catch is that we are still building the `n_days` list in memory